r/tests/testthat/test-expression.R

```diff
@@ -13,27 +13,54 @@ mock_scdata <- function() {
   return(pbmc_small)
 }
 
-test_that("Expression task returns appropiate number of cells and genes.", {
+test_that("Expression task returns appropiate number and names of genes.", {
   data <- mock_scdata()
   req <- mock_req()
 
   res <- runExpression(req, data)
+
   expect_equal(length(res), 2)
+  expect_equal(as.list(names(res)), req$body$genes)
+})
+
+test_that("Expression task returns appropiate number of cells.", {
+  data <- mock_scdata()
+  req <- mock_req()
+
+  res <- runExpression(req, data)
+  raw_expression <- res[[1]]$rawExpression$expression
+  truncated_expression <- res[[1]]$truncatedExpression$expression
 
-  expect_equal(nrow(res$rawExpression), nrow(res$truncatedExpression))
-  expect_equal(ncol(res$rawExpression), ncol(res$truncatedExpression))
 
-  expect_equal(nrow(res$rawExpression), ncol(data))
-  expect_equal(ncol(res$rawExpression), 2)
+  expect_equal(length(raw_expression), length(truncated_expression))
+
+  expect_equal(length(raw_expression), ncol(data))
 })
 
+test_that("Expression result is properly formated.", {
+  data <- mock_scdata()
+  req <- mock_req()
+
+  res <- runExpression(req, data)
+  raw_expression_result <- res[[1]]$rawExpression
+  truncated_expression_result <- res[[1]]$truncatedExpression
+  expect_true(!all(is.na(raw_expression_result$expression)))
+
+  expect_equal(raw_expression_result$mean, mean(raw_expression_result$expression, na.rm = TRUE))
+  expect_equal(raw_expression_result$stdev, sd(raw_expression_result$expression, na.rm = TRUE))
+  expect_equal(truncated_expression_result$min, min(truncated_expression_result$expression, na.rm = TRUE))
+  expect_equal(truncated_expression_result$max, max(truncated_expression_result$expression, na.rm = TRUE))
+})
+
+
+
 test_that("Expression task continues if gene doesnt exist and returns appropiate number of results", {
   data <- mock_scdata()
   req <- mock_req()
   req$body$genes <- append(req$body$genes, "aaa")
 
   res <- runExpression(req, data)
-  expect_equal(ncol(res$rawExpression), 2)
+  expect_equal(length(res), 2)
 })
 
 test_that("Expression task works with one gene", {
```

Comment:
what about a test whether it works with multiple genes?

Reply:
what about a test to check if the expression task is case sensitive? We had this problem a few months ago, where gene search was failing unless the user types in the gene using exact same letter cases.

Reply:
Addressed in a new ticket!